clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
@guozhangwang If subscribe(empty-list) is similar to pause, would you expect it to trigger a rebalance or hold onto the assigned partitions? If we expect to rebalance with an empty subscription, then what would be the advantage of staying in the group? The only thing I can think of is that it would lead to quicker reassignment than if we simply stopped sending heartbeats (supposing we don't have LeaveGroup). On the other hand, if we hold onto the partitions as in pause(), that would make SubscriptionState even trickier to manage, and would probably not be intuitive for the user.

Anyway, I'll update the patch with the original approach.